packages/design-system/src/components/slider/types.ts

```diff
@@ -0,0 +1,109 @@
+import type {
+  NumberFieldProps as AriaNumberFieldProps,
+  SliderRenderProps as RACSliderRenderProps,
+  SliderProps as RACSliderProps,
+  SliderThumbProps as RACSliderThumbProps,
+  SliderThumbRenderProps as RACSliderThumbRenderProps,
+  SliderTrackRenderProps as RACSliderTrackRenderProps,
+  SliderTrackProps as RACSliderTrackProps,
+  SliderOutputProps as RACSliderOutputProps,
+} from 'react-aria-components';
+import type { RenderPropsChildren } from '../../types';
+import type { PartialDeep } from 'type-fest';
+import type { InputClassNames, InputRenderProps } from '../input';
+import type { GroupClassNames } from '../group';
+
+export type SliderClassNames = PartialDeep<{
+  slider: {
+    container: string;
+    label: string;
+    min: string;
+    max: string;
+  };
+  track: {
+    container: string;
+    track: string; // TODO: adjust the styles to use this class
+  };
+  thumb: {
+    container: string;
+    thumb: string;
+  };
+  input: InputClassNames;
+  output: {
+    container: string;
+  };
+  group: GroupClassNames;
+}>;
+
+export type LabelAlignment = 'top' | 'left';
+
+export type SliderRenderProps = RACSliderRenderProps;
+export type SliderThumbRenderProps = RACSliderThumbRenderProps;
+export type SliderTrackRenderProps = RACSliderTrackRenderProps;
+
+type BaseSliderProps = {
+  children?: RenderPropsChildren<SliderRenderProps>;
+  label?: string;
+  classNames?: SliderClassNames;
+  alignLabel?: LabelAlignment;
+  includeTextField?: boolean;
```

Comment:
We shouldn't be having props for the `include*`. The determining factor for whether an element shows up or not is whether it was composed in or left out